### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Mellbourn** [oblador/loki#438]
I'm trying to get loki running in a react-native solution (based on [nx](https://nx.dev/)) that uses TypeScript.

My starting state is that I have a working Storybook. The http://localhost:7007 site is working and controlling the stories rendered in the ios emulator. 

So, first of all, I'm doing manual configuration because `loki init` wants to update a JavaScript file `storybook.js`, but I only have TypeScript files. But as far as I can see, the important things that `loki init` does is to add a config to the `package.json` and an `import 'loki/configure-react-native';` to the `storybook.js`, so I've inserted the latter manually into my `storybook.ts` file.

But after adding `import 'loki/configure-react-native';` , I get an infinite call loop on startup:
<img width="435" alt="Maximum call stack size exceeded" src="https://user-images.githubusercontent.com/1785727/220694623-09b8a6da-7e87-4cfe-b23b-fcd5476c2c67.png">

It seems to be specifically the `import 'loki/configure-react-native';` statement that is causing problems. If I remove it, everything works as usual.

**Comment by oblador** [comment]
What combo of RN and SB are you using? Are you using hermes as JS engine (I've had stack size issues with that one before)?

**Comment by Mellbourn** [comment]
Thanks for the quick reply Joel!

We're using hermes-engine 0.70.6

We've managed to fix the error by changing the setting `inlineRequires` to `false` in `metro.config.js`:

```JavaScript
  return withNxMetro(
    {
      transformer: {
        getTransformOptions: async () => ({
          transform: {
            experimentalImportSupport: false,
            inlineRequires: false,
          },
        }),
```

**Comment by oblador** [comment]
Yeah, what I found that was particularly troublesome with storybook and Hermes was the `core-js` polyfills. Since SB 6 for RN is really close to be stable, I think it's time that I update the example to use it (finally CSF! 😌) and see if I can reproduce the error (and hopefully fix).

**Comment by oblador** [comment]
Some early findings, I couldn't get anything basic working without applying this: https://github.com/storybookjs/react-native/issues/405#issuecomment-1436683333
